### PR TITLE
Add "Not yet supported" addon fragment 

### PR DESCRIPTION
**Pull request by gabrielluong**
This imports AC's [NotYetSupportedAddonActivity](https://github.com/mozilla-mobile/android-components/blob/ba0ab51d6de733c27d235e8863a9a5733654fd14/samples/browser/src/main/java/org/mozilla/samples/browser/addons/NotYetSupportedAddonActivity.kt) to Fenix.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.feature.addons.Addon
+import mozilla.components.feature.addons.ui.UnsupportedAddonsAdapter
+import mozilla.components.feature.addons.ui.UnsupportedAddonsAdapterDelegate
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.showToolbar
+
+/**
+ * Fragment for managing add-ons that are not yet supported by the browser.
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: nit: Fragment for displaying and managing ...

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_not_yet_supported_addons, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val context = requireContext()
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: nit: `view.context`?

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.mozac_feature_addons_unsupported_section))
+    }
+
+    override fun onUninstallError(addonId: String, throwable: Throwable) {
+        this@NotYetSupportedAddonFragment.view?.let { view ->
+            showSnackBar(view, "Failed to remove add-on")
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: nit: Hard-coded string... Can we move this to `feature-addons` (especially for localization reasons)?

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    override fun onUninstallSuccess() {
+        this@NotYetSupportedAddonFragment.view?.let { view ->
+            showSnackBar(view, "Successfully removed add-on")
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: nit: Hard-coded string... same as above.

**Comment by Amejia481** [comment]
@gabrielluong I pushed to update the strings with the ones AC

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt (hunk 4 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: Fixed!

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt (hunk 3 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: Fixed!

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: Fixed!

File: app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/addons/NotYetSupportedAddonFragment.kt]: Fixed!

**Comment by Amejia481** [comment]
@gabrielluong do you have access to Fire base? I'm not quite sure why this test keep failing `ui-test-x86-debug`

**Pull request merged**